extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/CommonPanacheQueryImpl.java

```diff
@@ -336,14 +335,9 @@ public <T extends Entity> T singleResult() {
 
     @SuppressWarnings("unchecked")
     public <T extends Entity> Optional<T> singleResultOptional() {
-        SelectionQuery hibernateQuery = createQuery(2);
```

Comment:
Interestingly, we were using a limit for `singleResultOptional`, but not for `singleResult` :|